src/bin/nydus-image/core/node.rs

```diff
@@ -1079,20 +1079,27 @@ impl Node {
     }
 
     /// Set node offset in bootstrap and return the next position.
-    pub(crate) fn set_v6_offset(&mut self, bootstrap_ctx: &mut BootstrapContext) {
+    pub(crate) fn set_v6_offset(
+        &mut self,
+        bootstrap_ctx: &mut BootstrapContext,
+        hardlink_v6_offset: u64,
```

Comment:
How about using `Option<u64>` here? So we could use `None` for non-hardlink files.

Reply:
Otherwise this comment, LGTM

Reply:
Done. How about this patch?